packages/@angular/cli/blueprints/component/index.ts

```diff
@@ -71,22 +72,25 @@ export default Blueprint.extend({
       type: Boolean,
       default: false,
       description: 'Specifies if declaring module exports the component.'
-    }
+    },
+    { name: 'app', type: String, aliases: ['a'] }
   ],
 
   beforeInstall: function (options: any) {
+    const appConfig = appUtils.getAppFromConfig(this.project.ngConfig.apps, this.options.app);
     if (options.module) {
       // Resolve path to module
       const modulePath = options.module.endsWith('.ts') ? options.module : `${options.module}.ts`;
-      const parsedPath = dynamicPathParser(this.project, modulePath);
+      const parsedPath = dynamicPathParser(this.project, modulePath, appConfig);
       this.pathToModule = path.join(this.project.root, parsedPath.dir, parsedPath.base);
 
       if (!fs.existsSync(this.pathToModule)) {
         throw 'Module specified does not exist';
       }
     } else {
       try {
-        this.pathToModule = findParentModule(this.project, this.dynamicPath.dir);
+        this.pathToModule = findParentModule
```

Comment:
Put as many arguments on the same line as function declaration, then indent 4 (or left align with first argument) the rest.